pkg/resourcemanager/controller/tokenrequestor/reconciler.go

```diff
@@ -94,24 +92,10 @@ func (r *reconciler) Reconcile(reconcileCtx context.Context, req reconcile.Reque
 		return reconcile.Result{}, fmt.Errorf("could not fetch Secret: %w", err)
 	}
 
-	if secret.DeletionTimestamp != nil || !isRelevantSecret(secret) {
-		if shouldDeleteServiceAccount(secret) {
-			if err := r.deleteServiceAccount(ctx, secret); err != nil {
-				return reconcile.Result{}, err
-			}
-		}
-
-		if err := controllerutils.PatchRemoveFinalizers(ctx, r.client, secret, finalizerName); err != nil {
-			return reconcile.Result{}, fmt.Errorf("error removing finalizer from Secret: %+v", err)
-		}
-
+	if !isRelevantSecret(secret) {
```

Comment:
Since we don't have a finalizer anymore, if such `Secret` is deleted then it disappears from the system immediately. Hence, when trying to read it then we get a "not found" error and exit early.